SpatialGDK/Source/SpatialGDK/Private/SpatialView/ViewDelta.cpp

```diff
@@ -77,83 +77,99 @@ const TArray<EntityComponentCompleteUpdate>& ViewDelta::GetCompleteUpdates() con
 	return EntityComponentChanges.GetCompleteUpdates();
 }
 
-TUniquePtr<AbstractOpList> ViewDelta::GenerateLegacyOpList() const
+const TArray<Worker_Op>& ViewDelta::GetWorkerMessages() const
 {
-	// Todo - refactor individual op creation to an oplist type.
-	TArray<Worker_Op> OpList;
-	OpList.Reserve(CreateEntityResponses.Num());
-
-	// todo Entity added ops get created here.
-
-	// todo Component Added ops get created here.
+	return WorkerMessages;
+}
 
-	for (const EntityComponentId& Id : AuthorityChanges.GetAuthorityLost())
-	{
-		Worker_Op Op = {};
-		Op.op_type = WORKER_OP_TYPE_AUTHORITY_CHANGE;
-		Op.op.authority_change.entity_id = Id.EntityId;
-		Op.op.authority_change.component_id = Id.ComponentId;
-		Op.op.authority_change.authority = WORKER_AUTHORITY_NOT_AUTHORITATIVE;
-		OpList.Push(Op);
-	}
+void ViewDelta::Clear()
+{
+	WorkerMessages.Empty();
+	OpLists.Empty();
+	AuthorityChanges.Clear();
+	EntityComponentChanges.Clear();
+	ConnectionStatus = 0;
+}
 
-	for (const EntityComponentId& Id : AuthorityChanges.GetAuthorityLostTemporarily())
+void ViewDelta::ProcessOp(const Worker_Op& Op, TSet<EntityComponentId>& ComponentsPresent)
+{
+	switch (static_cast<Worker_OpType>(Op.op_type))
 	{
-		Worker_Op Op = {};
-		Op.op_type = WORKER_OP_TYPE_AUTHORITY_CHANGE;
-		Op.op.authority_change.entity_id = Id.EntityId;
-		Op.op.authority_change.component_id = Id.ComponentId;
-		Op.op.authority_change.authority = WORKER_AUTHORITY_NOT_AUTHORITATIVE;
-		OpList.Push(Op);
+	case WORKER_OP_TYPE_DISCONNECT:
+		ConnectionStatus = Op.op.disconnect.connection_status_code;
+		DisconnectReason = FString(Op.op.disconnect.reason);
+		break;
+	case WORKER_OP_TYPE_FLAG_UPDATE:
+	case WORKER_OP_TYPE_LOG_MESSAGE:
+	case WORKER_OP_TYPE_METRICS:
+		WorkerMessages.Add(Op);
+		break;
+	case WORKER_OP_TYPE_CRITICAL_SECTION:
+		// Ignore.
+		break;
+	case WORKER_OP_TYPE_ADD_ENTITY:
+		EntityPresenceChanges.AddEntity(Op.op.add_entity.entity_id);
+		break;
+	case WORKER_OP_TYPE_REMOVE_ENTITY:
+		EntityPresenceChanges.AddEntity(Op.op.remove_entity.entity_id);
```

Comment:
AddEntity -> RemoveEntity?

Reply:
Oops :(. It's interesting to know that didn't break everything, although add and remove entity are largely vistigal so 🤷